### ISSUE TITLE
obs-studio.install failing with latest version of OBS (27.1.1)

### ISSUE THREAD

**Issue by sumo300** [AeliusSaionji/chocopkgs#36]
ERROR: Running ["C:\ProgramData\chocolatey\lib\obs-studio.install\tools\OBS-Studio-27.1.1-Full-Installer-x64_x64.exe" /S ] was not successful. Exit code was '6'. Exit code indicates the following: Setup process was forcefully terminated by the debugger..

After the failure, Chocolatey thinks the install succeeded as it indicates the package was installed.

Chocolatey is latest version being run from a PowerShell Core prompt (latest version of posh core).

```
2021-09-29 08:08:15,821 1316 [INFO ] - Installing 64-bit obs-studio.install...
2021-09-29 08:08:15,824 1316 [DEBUG] - Running Start-ChocolateyProcessAsAdmin -validExitCodes '0' -workingDirectory 'C:\ProgramData\chocolatey\lib\obs-studio.install\tools' -statements '/S ' -exeToRun 'C:\ProgramData\chocolatey\lib\obs-studio.install\tools\OBS-Studio-27.1.1-Full-Installer-x64_x64.exe' 
2021-09-29 08:08:15,824 1316 [DEBUG] - Test-ProcessAdminRights: returning True
2021-09-29 08:08:15,825 1316 [DEBUG] - Elevating permissions and running ["C:\ProgramData\chocolatey\lib\obs-studio.install\tools\OBS-Studio-27.1.1-Full-Installer-x64_x64.exe" /S ]. This may take a while, depending on the statements.
2021-09-29 08:08:18,386 1316 [DEBUG] - Command ["C:\ProgramData\chocolatey\lib\obs-studio.install\tools\OBS-Studio-27.1.1-Full-Installer-x64_x64.exe" /S ] exited with '6'.
2021-09-29 08:08:18,388 1316 [WARN ] - WARNING: Setup process was forcefully terminated by the debugger.
2021-09-29 08:08:18,583 1316 [ERROR] - ERROR: Running ["C:\ProgramData\chocolatey\lib\obs-studio.install\tools\OBS-Studio-27.1.1-Full-Installer-x64_x64.exe" /S ] was not successful. Exit code was '6'. Exit code indicates the following: Setup process was forcefully terminated by the debugger..
2021-09-29 08:08:18,586 1316 [DEBUG] - Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\obs-studio.install\tools\chocolateyinstall.ps1' -installArguments '' -packageParameters '''] exited with '6'.
```

**Comment by AeliusSaionji** [comment]
Exit code 6 happens when a video device is in use. You must exit slack / discord / ms teams / skype / or anything that might be accessing a video device before updating.

**Comment by AeliusSaionji** [comment]
Feel free to reopen this if the issue persists.

**Comment by Destroy666x** [comment]
I created a suggestion for OBS to introduce a wa to automate this if anyone else is quite annoyed by this: https://ideas.obsproject.com/posts/2107/installer-should-be-able-to-auto-kill-conflicting-processes Basically currently it's jsut better to download installer manually and upgrade there.

**Comment by AeliusSaionji** [comment]
You can use `--not-silent` to have chocolatey automate downloading and running the installer "normally".

**Comment by d-Rickyy-b** [comment]
I am also having this issue with the latest version 31.0.0. According to [this comment](https://community.chocolatey.org/packages/obs-studio.install#comment-6444990138), there are multiple other processes that block the installation. A list from that comment:

> - msedge
> - firefox
> - Skype
> - WebexHost
> - WhatsApp

But basically it comes down to any application that uses or can use the camera in one way or the other. The installer currently only checks for Teams and Slack. So maybe there need to be some more checks for other apps.

https://github.com/AeliusSaionji/chocopkgs/blob/4019a5db56b0b789d5f6c11aab31a6fd003bac50/obs-studio.install/tools/chocolateyinstall.ps1#L5-L12

**Comment by Destroy666x** [comment]
There's 0 point of checking apps. There can be hundreds of them. E.g. DeepL gets involved as well because of "Cefsharp subprocess" and it doesn't use camera. It's just shitty installer coding by OBS and once again, needs to be fixed by them.

**Comment by AeliusSaionji** [comment]
I agree it's largely pointless.

The checks are only there because someone recently wrote a PR to put them in.

Probably a better version of this PR should have captured the installer error code to say "please run with --not-silent".

Then the installer can tell the user why it's sad this time.

Or better yet, maybe a write-host "this package installer fails when the wind blows", just to set expectations :)

**Comment by d-Rickyy-b** [comment]
I agree that it doesn't make sense to check for a load of different apps. Another warn/error message would still be appreciated, if possible. The first two times I received the error message, I just ignored it and tried to install at a later point again. The third time I decided to see where the issue comes from and after some time I understood the reason for it.

If there had been an error message that tells me "Some application blocks the installation of obs" and maybe even a link to a wiki page, that explains what kind of apps could cause this, I guess my debugging would have ended a lot earlier 😄. Not sure if that is possible through chocolatey install scripts tho.
